### ISSUE TITLE
[OpenAPI] Property schema has been overwritten due to suspicious type check

### ISSUE THREAD

**Issue by c1tru55** [api-platform/core#6883]
**API Platform version(s) affected**: 4.0.12

**Description**  

in custom custom decorated `api_platform.metadata.property.metadata_factory` I return following property schema (`schema_type = input`):
```
  'exclusiveMinimum' => 0,
  'type' =>  [
    'integer',
    'null',
  ],
```
in a result on this line https://github.com/api-platform/core/blob/4.1/src/JsonSchema/SchemaFactory.php#L189 I have following variable values:
```
$isUnknown = false;
$propertySchema = [
  'exclusiveMinimum' => 0,
  'type' =>  [
    'integer',
    'null',
  ],
];
$propertySchemaType = [
  'integer',
  'null',
];
```
but since this condition is not passed:
```
        if (
            !$isUnknown && (
                [] === $types
                || ($propertySchema['$ref'] ?? $propertySchema['anyOf'] ?? $propertySchema['allOf'] ?? $propertySchema['oneOf'] ?? false)
                || (\is_array($propertySchemaType) ? \array_key_exists('string', $propertySchemaType) : 'string' !== $propertySchemaType)
                || ($propertySchema['format'] ?? $propertySchema['enum'] ?? false)
            )
        ) {
            $schema->getDefinitions()[$definitionName]['properties'][$normalizedPropertyName] = new \ArrayObject($propertySchema);

            return;
        }
```
my schema is changed to:
```
$propertySchema = [
  'exclusiveMinimum' => 0,
  'anyOf' => 
  [
    [
      '$ref' => '#/components/schemas/Entity.jsonld-Input',
    ],
    [
      'type' => 'null',
    ],
  ],
];
```
which is not I want.

**How to reproduce**  

**Possible Solution**  

not sure about this condition, why generated schema is changed at all: https://github.com/api-platform/core/blob/4.1/src/JsonSchema/SchemaFactory.php#L189
but this check looks suspicious:
```
\is_array($propertySchemaType) ? \array_key_exists('string', $propertySchemaType) : 'string' !== $propertySchemaType
```
especially when `type` in openapi schema should be string or plain array of strings.
https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-6.1.1

**Additional Context**

**Comment by soyuka** [comment]
To be sure are you hitting this line: 

https://github.com/api-platform/core/blob/a11988e50082fcca83b8d08b2bd6e4ab69dbd50c/src/JsonSchema/Metadata/Property/Factory/SchemaPropertyMetadataFactory.php#L111

I think that we're not detecting properly that you set the type yourself, can you try adding this flag to your `ApiProperty`:

https://github.com/api-platform/core/blob/9389b4f4637b925e55ee68c3f62c88828014c14f/src/JsonSchema/SchemaFactory.php#L167

**Comment by c1tru55** [comment]
> I think that we're not detecting properly that you set the type yourself, can you try adding this flag to your `ApiProperty` to hit this breakpoint:
> 
> https://github.com/api-platform/core/blob/9389b4f4637b925e55ee68c3f62c88828014c14f/src/JsonSchema/SchemaFactory.php#L167

yeah, came to same solution temporary.

**Comment by soyuka** [comment]
Agreed although you'd have the same issue no? To know why the check is there we probably need to comment + run the tests. 

When metadata is read from attributes (https://github.com/api-platform/core/blob/a8d0c8471060039bdd0b8d118dcf7f0e97709276/src/JsonSchema/Metadata/Property/Factory/SchemaPropertyMetadataFactory.php#L58) we  set this flag when the user defined the type, or else we compute it. As you're using metadata factories this check isn't added.
